airflow/models/abstractoperator.py

```diff
@@ -239,19 +242,27 @@ def global_operator_extra_link_dict(self) -> Dict[str, Any]:
     def extra_links(self) -> List[str]:
         return list(set(self.operator_extra_link_dict).union(self.global_operator_extra_link_dict))
 
-    def get_extra_links(self, dttm: datetime.datetime, link_name: str) -> Optional[Dict[str, Any]]:
+    def get_extra_links(self, ti: "TaskInstance", link_name: str) -> Optional[str]:
         """For an operator, gets the URLs that the ``extra_links`` entry points to.
 
         :raise ValueError: The error message of a ValueError will be passed on through to
             the fronted to show up as a tooltip on the disabled link.
-        :param dttm: The datetime parsed execution date for the URL being searched for.
+        :param ti: The TaskInstance for the URL being searched for.
         :param link_name: The name of the link we're looking for the URL for. Should be
             one of the options specified in ``extra_links``.
         """
-        if link_name in self.operator_extra_link_dict:
-            return self.operator_extra_link_dict[link_name].get_link(self, dttm)
-        elif link_name in self.global_operator_extra_link_dict:
-            return self.global_operator_extra_link_dict[link_name].get_link(self, dttm)
+        link: Optional["BaseOperatorLink"] = self.operator_extra_link_dict.get(
+            link_name
+        ) or self.global_operator_extra_link_dict.get(link_name)
```

Comment:
```suggestion
        link: Optional["BaseOperatorLink"] = (
            self.operator_extra_link_dict.get(link_name)
            or self.global_operator_extra_link_dict.get(link_name)
        )
```

Would Black be happy with this? The current format is… bad…

Reply:
No, sadly not :(

Reply:
I could do it like this:

```python
        link: Optional["BaseOperatorLink"] = self.operator_extra_link_dict.get(
            link_name, self.global_operator_extra_link_dict.get(link_name, None)
        )
```

(Which will do a unnecessary lookup to the global links)

Or this

```python
        link: Optional["BaseOperatorLink"] = self.operator_extra_link_dict.get(link_name)
        if not link:
            link = self.global_operator_extra_link_dict.get(link_name)
```

Reply:
Did the later in a43f10836